### ISSUE TITLE
increase efficiency for experimental evaluation

### ISSUE THREAD

**Issue by fmohr** [fmohr/asforests#3]
In the TMLR 2026 experiment setup, ground truth was computed separately for different validation set sizes, which is inefficient. However, this could not fixed after a large portion of experiments had already been run, because the training data is separated only *after* the validation set is split away, which means that the deviation matrices depend on the validation size. This is not ideal, because the deviation matrices should not depend on the validation size; it's not wrong but it induces possible inefficiencies since ground truth has to be computed repeatedly for reproducibility (without any need).
